actions/upload_sessions.go

```diff
@@ -358,3 +361,24 @@ func (usr *UploadSessionResource) GetPaymentStatus(c buffalo.Context) error {
 func sqlWhereForGenesisHashAndChunkIdx(genesisHash string, chunkIdx int) string {
 	return fmt.Sprintf("(genesis_hash = '%s' AND chunk_idx = %d)", genesisHash, chunkIdx)
 }
+
+func alphaWaitForTransfer(ethAddr string, uploadSessionId string) {
+	success := services.EthWrapper.WaitForTransfer(services.StringToAddress(ethAddr))
+
+	session := models.UploadSession{}
+	if err := models.DB.Find(&session, uploadSessionId); err != nil {
+		return
+	}
+
+	if success {
+		session.PaymentStatus = models.PaymentStatusConfirmed
+	} else {
+		session.PaymentStatus = models.PaymentStatusError
+	}
+
+	if err := models.DB.Save(&session); err != nil {
+		return
+	}
+
+	// TODO(pzhao5): NotifyBeta on payment status
+}
```

Comment:
I tend to agree with what Aaron mentioned in the other PR.  This method could just be called waitForTransfer and we could call it from both Create and Createbeta.